android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java

```diff
@@ -241,6 +243,33 @@ public void remove(ReadableArray tracks, final Promise callback) {
         });
     }
 
+    @ReactMethod
+    public void updateMetadata(String id, ReadableMap map, final Promise callback) {
+        waitForConnection(() -> {
+            ExoPlayback playback = binder.getPlayback();
+            List<Track> queue = playback.getQueue();
+            Track track = null;
+            int index = -1;
+
+            for(int i = 0; i < queue.size(); i++) {
+                track = queue.get(i);
+
+                if(track.id.equals(id)) {
+                    index = i;
+                    break;
+                }
+            }
+
+            if(index == -1) {
+                callback.reject("track_not_in_queue", "No track found");
+            } else {
+                track.setMetadata(getReactApplicationContext(), Arguments.toBundle(map), binder.getRatingType());
```

Comment:
`Track` has a property called: `originalItem` could you please make sure that these properties get merged into the `originalItem` please. You can see that I did that in iOS.

Reply:
`setMetadata` already deals with that, see lines 112 to 113 from `Track.java`

Reply:
Oops! I must have missed that 😭